api/src/main/java/org/openmrs/module/fgh/mpi/MpiHttpClient.java

```diff
@@ -68,26 +88,52 @@ public Map<String, Object> getPatient(String patientUuid) throws Exception {
 			log.debug("Searching for patient from MPI with OpenMRS uuid: " + patientUuid);
 		}
 		
+		String mpiSystem = getGlobalPropertyValue(GP_MPI_SYSTEM);
+		
 		if (openmrsUuidSystem == null) {
 			synchronized (MpiHttpClient.class) {
 				openmrsUuidSystem = MpiUtils.getGlobalPropertyValue(GP_UUID_SYSTEM);
 			}
 		}
 		
 		String query = REQ_PARAM_SOURCE_ID + "=" + openmrsUuidSystem + "|" + patientUuid;
-		Map<String, Object> pixResponse = submitRequest(SUBPATH_PATIENT + "/$ihe-pix?" + query, null, Map.class);
+		
+		Map<String, Object> pixResponse = null;
+		
+		try {
+			pixResponse = submitRequest(SUBPATH_PATIENT + "/$ihe-pix?" + query, null, Map.class);
+		}
+		catch (APIException e) {
+			//When a queried patient is not present in Sante DB it throws an Exception.
```

Comment:
This comment is vague, sante can't be throwing an error that you are catching here, it's our code that must be throwing the error inside `submitRequest`

Reply:
Addressed